### PR TITLE
Test non-empty StorageIDs

### LINKED ISSUE #8697: Handle non-empty StorageID in unit-tests
Currently, only empty StorageIDs appears in the unit-tests.
These should also be tested on non-empty StorageIDs, to make sure the flows work.

### PR DESCRIPTION
**Pull request by itaigilo**
Closes #8697.

## Change Description

In order to provide better coverage:

1. Changing some of the unit-tests to run with a non-empty StorageID.
2. Replace empty StorageIDs with `config.SingleBlockstoreID`.

### CODE REVIEW

**Review by nadavsteindler — APPROVED** [review]
Looks good

File: pkg/graveler/committed/manager_test.go [hunk 1]
```diff
@@ -35,16 +36,16 @@ func TestManager_WriteRange(t *testing.T) {
 	}{
 		{
 			name:           "iterator_exhausted",
-			initStorageID:  "",
-			writeStorageID: "",
+			initStorageID:  config.SingleBlockstoreID,
```
Comment (arielshaqed) [inline pkg/graveler/committed/manager_test.go:39]: I don't understand what we did in this test, but also what we did in the other test files.  I would have _expected_ to have twice as many test cases: once for "" aka SIngleBlockstoreID, one for "sid1".  (This could have happened with nested `test.Run()`s, of course.)  But instead some test cases switch to testing "", others to "sid1".  And I don't understand why.

Obviously I don't understand this for all changes in this file, not only for this one.

**Review by arielshaqed — CHANGES_REQUESTED** [review]
THANKS!  Much needed unit tests.  But we might need more; these might be in a future PR, but then this PR has to say that and we need a more specific issue.

- Not sure how this tests CreateRepository.
- Don't we need to update other tests?  E.g. graveler_v2_test.go appears to pass empty storage IDs; there may be others.

Blocking on these 3 issues (including the one on catalog_test.go).  Entirely possible I misunderstood, and will happily unblock.

File: pkg/graveler/committed/manager_test.go (hunk 1 above)
Comment (itaigilo) [reply pkg/graveler/committed/manager_test.go:39]: As far as I can understand, we're trying to make sure `StorageID` flows properly from start to finish. By starting with either "" or "sid*" on one side, and making sure we're getting the same result in the other end, we have some confidence it's being passed properly, no?

File: pkg/graveler/testutil/graveler_mock.go [hunk 2]
```diff
@@ -37,3 +39,9 @@ func InitGravelerTest(t *testing.T) *GravelerTest {
 
 	return test
 }
+
+// ShortenBranchUpdateBackOff upgrade graveler branch update back-off to shorten test duration
```
Comment (itaigilo) [inline pkg/graveler/testutil/graveler_mock.go:43]: Found out the graveler tests were way too slow - 
Ended up adding this function to this PR.
Probably belongs to a separate one, but I think we can handle this.

(Anyway, this cuts running time from 90 seconds to only 8.)

**Comment by itaigilo** [comment]
> * CreateRepository

Thanks @arielshaqed fro your review!

1. graveler.CreateRepository - Is this tested anywhere? If so, I couldn't find any reference for it...
2. v2 - Right! I've missed that. Added non-empty StorageIDs there. Not sure it's enough, but the price of adding a new repo to these tests might be high (in terms of dev time, I guess).

File: pkg/graveler/committed/manager_test.go (hunk 1 above)
Comment (arielshaqed) [reply pkg/graveler/committed/manager_test.go:39]: I guess? To be convinced, I need to dig through the code and implementations and see that all interesting places are covered. 
I would have preferred to test everything twice, once with no storage id and once with. Then I wouldn't need to think. 
Not blocking.

**Review by arielshaqed — APPROVED** [review]
I am approving because I would rather have _some_ tests than none, and this does add tests.  But these tests are less comprehensive than the title "test non-empty StorageIDs" suggests to literal-minded people like me.

> graveler.CreateRepository - Is this tested anywhere? If so, I couldn't find any reference for it...

```sh
❯ git grep -l CreateRepository pkg/ | fgrep _test.go
pkg/api/controller_test.go
pkg/catalog/fake_graveler_test.go
pkg/graveler/ref/branch_iterator_test.go
pkg/graveler/ref/manager_test.go
pkg/graveler/ref/pulls_iterator_test.go
pkg/graveler/ref/repository_iterator_test.go
pkg/graveler/ref/resolve_ref_test.go
pkg/graveler/ref/tag_iterator_test.go
```

Of these, I reckon controller_test.go does the most testing.  But regardless, the fact that CreateRepository works _without_ the storage ID change does not mean that it works _with_ it.  So a change must be accompanied by tests.  Sometimes this will mean engineering a better test.

File: pkg/graveler/committed/manager_test.go (hunk 1 above)
Comment (itaigilo) [reply pkg/graveler/committed/manager_test.go:39]: Makes sense.
Added.

**Review by itaigilo — COMMENTED** [review]
Thanks @arielshaqed for your comments, and for striving for making our testing better.

I've turned some of the suggested tests to use non-empty StorageIDs,
But as you wrote, most of the logic is being tested in the controller tests.
These are tricky to test with a non-empty StorageID, because it currently validates that the StorageID _is_ empty.

The internal parts should cover some of this, while Esti tests (in other projects) should also provide more coverage for this.

I'm merging this PR, and we are still working on adding more Esti tests that will cover this, but if you still feel there are more unit-tests we can add to increase our present and future confidence, I'd be happy to talk about it.
